### DISCUSSION TITLE
Environment variable assistance with ImageMagick

### DISCUSSION THREAD

**Discussion by hokiecolt** [ImageMagick/ImageMagick#5997] [Help]
### ImageMagick version

7.1.0-16

### Operating system, version and so on

Oracle Enterprise Linux 7

### Description

Good afternoon everyone!  
I'm new to ImageMagick.  We are using Alfresco and want to utilize the Transform Service.  We have installed 7.1.0-16 using the RPMs.  When we run magick  --version we get the correct response.  Here's where I need some help.  I'm trying to figure out what the IMAGEMAGICK_EXE, IMAGEMAGICK_CONF and IMAGEMAGICK_DYN environment variables should be set too.  I I know IMAGEMAGICK_HOME is /usr/lib64/ImageMagick7.1.0  but there is no lib or bin directories under that folder so I'm just trying to figure out what I need to use for this values.  Any assistance is appreciated!  
Thanks for your time!

Best regards,
Steve

**Comment by snibgo** [comment]
Where did you find references to environment variables IMAGEMAGICK_CONF etc? I can't find them in ImageMagick source or documentation. Perhaps they are features of Alfresco.

**Comment by hokiecolt** [comment]
Snibgo,

 

Good afternoon!

 

I think you might be right.  I was looking on the ImageMagick site and finally found the page that has the environment variable definitions.  So I’ve also reached out to Alfresco to see what they need for that environment variable if ImageMagick is installed by RPM.  I’ll close my topic.

 

Thanks for replying!  Appreciate it!

 

Best regards,

 

Steve

 

From: snibgo ***@***.***> 
Sent: Monday, January 23, 2023 2:19 PM
To: ImageMagick/ImageMagick ***@***.***>
Cc: Boyd, Steven R CTR (USA) ***@***.***>; Author ***@***.***>
Subject: [URL Verdict: Neutral][Non-DoD Source] Re: [ImageMagick/ImageMagick] Environment variable assistance with ImageMagick (Discussion #5997)

 

All active links contained in this email were disabled. Please verify the identity of the sender, and confirm the authenticity of all links contained within the message prior to copying and pasting the address to a Web browser.

 

  _____  

 

Where did you find references to environment variables IMAGEMAGICK_CONF etc? I can't find them in ImageMagick source or documentation. Perhaps they are features of Alfresco.

**Comment by hokiecolt** [comment]
Snibgo,

Good morning!

Sorry to bug you (again).  As my post said, we've installed ImageMagick using the rpms.  It creates a folder /usr/lib64/ImageMagick7.1.0. Where are the binaries?  Is there a /lib or /bin that's associated with ImageMagick?  The folks at Alfresco have the following posted on their website:


To enable image manipulation in Transform Service, you must install and configure ImageMagick. Transform Service uses ImageMagick to manipulate images for previewing.

  1.  Check if ImageMagick is already installed on your system. Use the ImageMagick convert command to check that you have the right software installed on your machine. This command is usually located in /usr/bin: install Image.
  2.  If the ImageMagick software isn’t available on your system, download and install the appropriate package for your platform. To download ImageMagick, browse to ImageMagick download website<https://www.imagemagick.org/script/download.php>.

Note: In next steps, you’ll make changes to the Content Services configuration files to enable the manually installed ImageMagick application. These steps can only be performed after Content Services has been installed.

The following table lists example of how to set the paths to different things when starting Transform Core AIO later on:

Property        Description
img.root        Windows: img.root=C:\\ImageMagick
Linux: img.root=/ImageMagick

Note: Don’t include a slash (/) at the end of the path, i.e. /ImageMagick/.
img.dyn Windows: img.dyn=${img.root}\\lib
Linux: img.dyn=${img.root}/lib
img.exe Windows: img.exe=${img.root}\\convert.exe
Linux: img.exe=${img.root}/bin/convert
img.coders      Windows: img.coders=${img.root}\\modules\\coders
Linux: img.coders=${img.root}/modules/coders
img.config      Windows: img.config=${img.root}\\config
Linux: img.config=${img.root}/config
img.url Windows: img.url=${img.root}\\url
Linux: img.url=${img.root}/url
So my question is other than /usr/lib64/ImageMagick7.1.0 where are the files installed?

Any help is appreciated.

Best rgards,

Steve
________________________________
From: snibgo ***@***.***>
Sent: Monday, January 23, 2023 2:19 PM
To: ImageMagick/ImageMagick ***@***.***>
Cc: Boyd, Steven R CTR (USA) ***@***.***>; Author ***@***.***>
Subject: [URL Verdict: Neutral][Non-DoD Source] Re: [ImageMagick/ImageMagick] Environment variable assistance with ImageMagick (Discussion #5997)


All active links contained in this email were disabled. Please verify the identity of the sender, and confirm the authenticity of all links contained within the message prior to copying and pasting the address to a Web browser.


________________________________



Where did you find references to environment variables IMAGEMAGICK_CONF etc? I can't find them in ImageMagick source or documentation. Perhaps they are features of Alfresco.

**Comment by snibgo** [comment]
> It creates a folder /usr/lib64/ImageMagick7.1.0. Where are the binaries?

I don't know where the binaries are on your computer. `ls` should tell you. Perhaps they are in that folder, or a `bin` folder within that folder, or elsewhere.

What happens when you type "convert" or "magick" at the command line? If ImageMagick runs, then it must be on your system path. Then:
```
whereis convert
```
... or ...
```
whereis magick
```
... will give the locations of those programs.

I don't know what those other directories are, or what software uses them. For example, a `/url` directory means nothing to me.

`/config` and `/coders` are used in the directory tree of the IM source code.  `/coders` is also a bottom-level directory in the directory tree of a built system.

**Comment by fmw42** [comment]
On linux, the usual place is /usr/local/bin or /usr/bin.  You should be able to find it using

`type convert`

or

`which convert`

If you are using IM 7, then 

`type magick`

or

`which magick`
